### ISSUE TITLE
CMake: No COMPONENT, No LIBRARY DESTINATION

### ISSUE THREAD

**Issue by tbiedert** [ospray/ospray#123]
With the latest update my application and custom module can not be built anymore due to CMake issues.

Relevant piece from CMakeLists.txt:
```
OSPRAY_CREATE_LIBRARY(ospray_module_ParticleSurface ParticleSurface.cpp ParticleSurface.ispc LINK ${OSPRAY_LIBRARIES})

OSPRAY_ADD_EXECUTABLE(app main.cpp)
OSPRAY_EXE_LINK_LIBRARIES(app ospray_module_ParticleSurface ${OSPRAY_LIBRARIES})
```

CMake output:
```
CMake Warning at /usr/local/lib/cmake/ospray-1.2.0/macros.cmake:181 (MESSAGE):
  No COMPONENT for installation specified!
Call Stack (most recent call first):
  /usr/local/lib/cmake/ospray-1.2.0/macros.cmake:211 (OSPRAY_SPLIT_CREATE_ARGS)
  CMakeLists.txt:31 (OSPRAY_CREATE_LIBRARY)


CMake Error at /usr/local/lib/cmake/ospray-1.2.0/macros.cmake:136 (INSTALL):
  install TARGETS given no LIBRARY DESTINATION for shared library target
  "ospray_module_ParticleSurface".
Call Stack (most recent call first):
  /usr/local/lib/cmake/ospray-1.2.0/macros.cmake:219 (OSPRAY_INSTALL_LIBRARY)
  CMakeLists.txt:31 (OSPRAY_CREATE_LIBRARY)


CMake Error at CMakeLists.txt:33 (OSPRAY_ADD_EXECUTABLE):
  Unknown CMake command "OSPRAY_ADD_EXECUTABLE".
```

If I replace the now missing OSPRAY_ADD_EXECUTABLE with OSPRAY_CREATE_APPLICATION, CMake will also complain about the missing component and library destination for the application target.

Thanks!

**Comment by johguenther** [comment]
The `OSPRAY_CREATE_*` convenience macros now also accept a COMPONENT parameter to specify the installation component, issuing a warning when this is missing. You can also set a default component via the variable `OSPRAY_DEFAULT_COMPONENT`. See also the documentation of the macros in [`macros.cmake` line 194ff](https://github.com/ospray/ospray/blob/master/components/ospcommon/cmake/macros.cmake#L194-L208).

Those macros also depend on `CMAKE_INSTALL_LIBDIR` and `CMAKE_INSTALL_BINDIR`set, to do so you could just include `GNUInstallDirs`, what OSPRay does in `ospray_options.cmake`.

**Comment by jeffamstutz** [comment]
This has been fixed in the latest v1.3.1 release. Please re-open if you continue to see the issue.
